src/Appwrite/Platform/Tasks/Upgrade.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Appwrite\Platform\Tasks;
+
+use Utopia\CLI\Console;
+use Utopia\Validator\Text;
+
+class Upgrade extends Install
+{
+    public static function getName(): string
+    {
+        return 'upgrade';
+    }
+
+    public function __construct()
+    {
+        $this
+            ->desc('Upgrade Appwrite')
+            ->param('httpPort', '', new Text(4), 'Server HTTP port', true)
+            ->param('httpsPort', '', new Text(4), 'Server HTTPS port', true)
+            ->param('organization', 'appwrite', new Text(0), 'Docker Registry organization', true)
+            ->param('image', 'appwrite', new Text(0), 'Main appwrite docker image', true)
+            ->param('interactive', 'Y', new Text(1), 'Run an interactive session', true)
+            ->callback(fn ($httpPort, $httpsPort, $organization, $image, $interactive) => $this->action($httpPort, $httpsPort, $organization, $image, $interactive));
+    }
+
+    public function action(string $httpPort, string $httpsPort, string $organization, string $image, string $interactive): void
+    {
+        // Check for previous installation
+        $data = @file_get_contents($this->path . '/docker-compose.yml');
+        if (empty($data)) {
+            Console::error('Appwrite installation not found.');
+            Console::log('The command was not run in the parent folder of your appwrite installation.');
+            Console::log('Please navigate to the parent directory of the Appwrite installation and try again.');
+            Console::log('  parent_directory <= you run the command in this directory');
+            Console::log('  └── appwrite');
+            Console::log('      └── docker-compose.yml');
+            Console::exit(1);
+        }
+        parent::action($httpPort, $httpsPort, $organization, $image, $interactive);
+    }
```

Comment:
It could also be a good idea to drop the migration command and initiate the migration directly here. We could add a confirmation dialog.

Reply:
We should keep the migrate command the same since there can be scenarios where you only need to run the migrate (for example, if you manually upgrade by modifying the docker-compose.yml file).

I also prefer to avoid automatically running the migration after an upgrade. I think it would be better to focus on a specific thing (the upgrade should be focused on upgrading the docker-compose.yml and .env files and migrate on migrating the db schema). Long run, I think it would be best to decouple migrations from the upgrade even more so that you don't need to run the migration after upgrading, and maybe you can do them in pieces per project or something.

On another note, I hate how the install command automatically starts appwrite because the docker compose context is in the container rather than on the host machine.